src/lucky/route_inferrer.cr

```diff
@@ -0,0 +1,99 @@
+class Lucky::RouteInferrer
```

Comment:
This class was copied from https://github.com/luckyframework/lucky/blob/master/src/run_macros/infer_route.cr and lightly modified